### PR TITLE
GetUsersByIds logic

### PR DESCRIPTION
**Pull request by lalagvana**

### CODE REVIEW

File: src/UserService.Business/GetUsersByIdsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            this.mapper = mapper;
+        }
+
+        public List<User> Execute(IEnumerable<Guid> usersIds)
+        {
+            var users = new List<User>();
+
+            foreach(Guid userId in usersIds)
+            {
+                users.Add(mapper.Map(repository.GetUserInfoById(userId)));
```
Comment (DarkFelas) [inline src/UserService.Business/GetUsersByIdsCommand.cs:31]: not rly good, this logic will create a lot of independent queries to our DB, plz create new method in user repository to get IEnumerable<DbUser> in single query

File: src/UserService.Data/UserRepository.cs [hunk 2]
```diff
@@ -38,13 +38,13 @@ public Guid CreateUser(DbUser dbUser, DbUserCredentials dbUserCredentials)
 
         public DbUser GetUserInfoById(Guid userId)
             => _provider.Users.FirstOrDefault(dbUser => dbUser.Id == userId) ??
-               throw new Exception("User with this id not found.");
+               throw new NotFoundException("User with this id was not found.");
```
Comment (DarkFelas) [inline src/UserService.Data/UserRepository.cs:41]: Add ID value to message

File: src/UserService.Data/UserRepository.cs [hunk 3]
```diff
@@ -38,13 +38,13 @@ public Guid CreateUser(DbUser dbUser, DbUserCredentials dbUserCredentials)
...
 
         public bool EditUser(DbUser user)
         {
             if (!_provider.Users.Any(users => user.Id == users.Id))
             {
-                throw new Exception("User was not found.");
+                throw new NotFoundException("User was not found.");
```
Comment (DarkFelas) [inline src/UserService.Data/UserRepository.cs:47]: Add ID value to message

File: src/UserService.Data/UserRepository.cs [hunk 4]
```diff
@@ -59,7 +59,7 @@ public DbUser GetUserByEmail(string userEmail)
 
             if (dbUser == null)
             {
-                throw new Exception("User credentials not found.");
+                throw new NotFoundException("User credentials not found.");
```
Comment (DarkFelas) [inline src/UserService.Data/UserRepository.cs:62]: Add email value to message

**Review by DarkFelas — CHANGES_REQUESTED** [review]

**Review by DarkFelas — APPROVED** [review]

File: test/UserService.Business.UnitTests/GetUsersByIdsCommandTests.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        private Mock<IUserRepository> repositoryMock;
+        private Mock<IMapper<DbUser, User>> mapperMock;
+
+        private Guid userId;
+        private List<Guid> usersIdsList;
+        private User user;
+        private List<User> usersList;
+        private DbUser dbUser;
+
+        [SetUp]
```
Comment (djino97) [inline test/UserService.Business.UnitTests/GetUsersByIdsCommandTests.cs:27]: OneTimeSetUp

File: test/UserService.Data.UnitTests/UserRepositoryTests.cs [hunk 6]
```diff
@@ -245,6 +246,30 @@ public void ShouldThrowExceptionWhenEmailIsAlreadyTaken()
...
+        }
+
+        [Test]
+        public void ShouldReturnDbUsersList()
+        {
+            var result = repository.GetUsersByIds(new List<Guid>() { firstUserId });
+
+            Assert.IsInstanceOf<IEnumerable<DbUser>>(result);
+            Assert.That(provider.Users, Is.EquivalentTo(new[] { firstUser }));
+            Assert.That(result, Is.EquivalentTo(new[] { firstUser }));
```
Comment (djino97) [inline test/UserService.Data.UnitTests/UserRepositoryTests.cs:269]: AreEqual?

**Review by djino97 — APPROVED** [review]

**Pull request merged**